File: app/src/main/java/com/blokkok/app/HomeFragment.java
package com.blokkok.app;

import android.content.Intent;
import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

/**
 * A simple {@link Fragment} subclass.
 * Use the {@link HomeFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class HomeFragment extends Fragment {
    public HomeFragment() {
        // Required empty public constructor
    }

    public static HomeFragment newInstance() {
        HomeFragment fragment = new HomeFragment();
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View viewInflater = inflater.inflate(R.layout.fragment_home, container, false);

        FloatingActionButton newProjectFab = viewInflater.findViewById(R.id.newProject);

        newProjectFab.setOnClickListener(v -> {
            Intent i1 = new Intent();

            i1.setClass(getActivity(), EditorActivity.class);
            startActivity(i1);
            getActivity().finish();
        });

        return viewInflater;
    }
}

File: app/src/main/java/com/blokkok/app/HomeFragment.kt
package com.blokkok.app

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.google.android.material.floatingactionbutton.FloatingActionButton

class HomeFragment : Fragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val viewInflater = inflater.inflate(R.layout.fragment_home, container, false)
        val newProjectFab: FloatingActionButton = viewInflater.findViewById(R.id.newProject)
        newProjectFab.setOnClickListener { v: View? ->
            val intent = Intent()
            intent.setClass(requireActivity(), EditorActivity::class.java)
            startActivity(intent)
            requireActivity().finish()
        }
        return viewInflater
    }

    companion object {
        fun newInstance(): HomeFragment {
            return HomeFragment()
        }
    }
}